homeassistant/components/husqvarna_automower/calendar.py

```diff
@@ -0,0 +1,114 @@
+"""Creates a calendar entity for the mower."""
+
+from datetime import datetime
+import logging
+
+from aioautomower.model import AutomowerCalendarEvent
+from ical.calendar import Calendar
+from ical.event import Event
+from ical.types.recur import Recur
```

Comment:
I saw that `aioautomower` already depends directly on `ical` and I think we may want to move the use of `ical` into `aioautomower` similar to how rainbird works. (Local calendar and local todo directly use ical because there is no other library that implements a local calendar or local todo)

Right now the library returns an rrule string so that a caller can do recurring event expansion, but I think it makes sense to push down recurring event expansion to the library rather than having home assistant do it.  Really our only actual agreed upon requirement is that the integration does the expansion, so its fine from that point of view, but my hunch is we're spreading around responsibilities a bit too much here.  I'm open to another opinion on this.

Reply:
I've copied a lot of code from pyrainbird and rainbrid and implemented it here. I haven't bumped the library separately, just in case other changes are requested.
The source code of the library can be found here: https://github.com/Thomas55555/aioautomower/tree/2024.9.0b1
It's not completely ready (dostrings, etc. are missing), but I will to this before the final release.